src/sync/sync_session.hpp

```diff
@@ -275,6 +275,7 @@ class SyncSession : public std::enable_shared_from_this<SyncSession> {
     size_t m_death_count = 0;
 
     SyncConfig m_config;
+    util::Optional<std::array<char, 64>> m_realm_encryption_key;
```

Comment:
Yeah, I was exploring both options, but it feels weird whichever way I try it. Okay, I'll move it to `SyncConfig`.